addons/sale/models/sale_order.py

```diff
@@ -1221,7 +1221,7 @@ def _create_invoices(self, grouped=False, final=False, date=None):
     def message_post(self, **kwargs):
         if self.env.context.get('mark_so_as_sent'):
             self.filtered(lambda o: o.state == 'draft').with_context(tracking_disable=True).write({'state': 'sent'})
-        return super(SaleOrder, self.with_context(mail_post_autofollow=self.env.context.get('mail_post_autofollow', True))).message_post(**kwargs)
+        return super(SaleOrder, self.with_context(mail_post_autofollow=self.env.context.get('mail_post_autofollow', True), lang=self.partner_id.lang)).message_post(**kwargs)
```

Comment:
This line is getting very long, maybe worth splitting it
```suggestion
        return super(SaleOrder, self.with_context(
            mail_post_autofollow=self.env.context.get('mail_post_autofollow', True),
            lang=self.partner_id.lang,
        )).message_post(**kwargs)
```